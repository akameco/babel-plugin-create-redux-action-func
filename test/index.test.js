// @flow
import transformCode from './transform'

test('basic', () => {
  const input = `
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type Action = { +type: HOGE_TYPE }
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('with args', () => {
  const input = `
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type Action = { +type: HOGE_TYPE, id: number }
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('two func with args', () => {
  const input = `
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type HOGE_FOO_TYPE = 'hoge/FOO_ACTION'
    type Action = { +type: HOGE_TYPE, +id: number } | { +type: HOGE_FOO_TYPE }
  `
  expect(transformCode(input)).toMatchSnapshot()
})
