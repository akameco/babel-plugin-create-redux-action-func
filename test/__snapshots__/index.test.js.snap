// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"// @flow
import type { Action } from './actionTypes';
import { HOGE } from './constants';

export function hoge(): Action {
    return {
        type: HOGE
    };
}"
`;

exports[`two func with args 1`] = `
"// @flow
import type { Action } from './actionTypes';
import { HOGE, HOGE_FOO } from './constants';

export function hoge(id: number): Action {
    return {
        type: HOGE,
        id
    };
}
export function hogeFoo(): Action {
    return {
        type: HOGE_FOO
    };
}"
`;

exports[`with args 1`] = `
"// @flow
import type { Action } from './actionTypes';
import { HOGE } from './constants';

export function hoge(id: number): Action {
    return {
        type: HOGE,
        id
    };
}"
`;
